latest libjxl source code fails to compile for aarch64-linux with:  ../jxl/modular/transform/palette.h:124:20: error: variable ‘std::atomic<int> num_errors’ has initializer but incomplete type

compiling latest git of libjxl for aarch64 is currently broken with: 

```
In file included from /usr/aarch64-unknown-linux-gnu/tmp/portage/media-libs/libjxl-9999/work/libjxl-9999/lib/jxl/modular/transform/transform.cc:11:
/usr/aarch64-unknown-linux-gnu/tmp/portage/media-libs/libjxl-9999/work/libjxl-9999/lib/jxl/modular/transform/palette.h: In function ‘jxl::Status jxl::InvPalette(jxl::Image&, uint32_t, uint32_t, uint32_t, jxl::Predictor, const jxl::weighted::Header&, jxl::ThreadPool*)’:
/usr/aarch64-unknown-linux-gnu/tmp/portage/media-libs/libjxl-9999/work/libjxl-9999/lib/jxl/modular/transform/palette.h:124:20: error: variable ‘std::atomic<int> num_errors’ has initializer but incomplete type
  124 |   std::atomic<int> num_errors{0};
      |                    ^~~~~~~~~~
``` 

to reproduce download latest git sources and compile libjxl for aarch64

**Environment**
 - OS: Linux
 - Compiler version: gcc-10.3.1
 - CPU type: cross-compile from amd64 to aarch64
 - cjxl/djxl version string: latest git snapshot

**Additional context**
this is cross compile, but I don't see anything cross-compile specific about the error. 

the compressed build log is attached: [build.log.gz](https://github.com/libjxl/libjxl/files/8661800/build.log.gz)

--- a/lib/jxl/dec_cache.h
+++ b/lib/jxl/dec_cache.h
@@ -8,6 +8,7 @@
 
 #include <stdint.h>
 
+#include <atomic>
 #include <hwy/base.h>  // HWY_ALIGN_MAX
 
 #include "jxl/decode.h"

--- a/lib/jxl/dec_modular.cc
+++ b/lib/jxl/dec_modular.cc
@@ -7,6 +7,7 @@
 
 #include <stdint.h>
 
+#include <atomic>
 #include <vector>
 
 #include "lib/jxl/frame_header.h"

--- a/lib/jxl/enc_image_bundle.cc
+++ b/lib/jxl/enc_image_bundle.cc
@@ -5,6 +5,7 @@
 
 #include "lib/jxl/enc_image_bundle.h"
 
+#include <atomic>
 #include <limits>
 #include <utility>
 

--- a/lib/jxl/enc_modular.cc
+++ b/lib/jxl/enc_modular.cc
@@ -9,6 +9,7 @@
 #include <stdint.h>
 
 #include <array>
+#include <atomic>
 #include <limits>
 #include <queue>
 #include <utility>

--- a/lib/jxl/enc_patch_dictionary.cc
+++ b/lib/jxl/enc_patch_dictionary.cc
@@ -10,6 +10,7 @@
 #include <sys/types.h>
 
 #include <algorithm>
+#include <atomic>
 #include <string>
 #include <tuple>
 #include <utility>

--- a/lib/jxl/modular/transform/palette.h
+++ b/lib/jxl/modular/transform/palette.h
@@ -6,6 +6,8 @@
 #ifndef LIB_JXL_MODULAR_TRANSFORM_PALETTE_H_
 #define LIB_JXL_MODULAR_TRANSFORM_PALETTE_H_
 
+#include <atomic>
+
 #include "lib/jxl/base/data_parallel.h"
 #include "lib/jxl/base/status.h"
 #include "lib/jxl/common.h"

--- a/lib/threads/thread_parallel_runner_test.cc
+++ b/lib/threads/thread_parallel_runner_test.cc
@@ -3,6 +3,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
+#include <atomic>
+
 #include "gtest/gtest.h"
 #include "lib/jxl/base/data_parallel.h"
 #include "lib/jxl/base/thread_pool_internal.h"

